packages/commonwealth/server/passport/magic.ts

```diff
@@ -7,367 +7,519 @@ import { Magic, MagicUserMetadata } from '@magic-sdk/admin';
-            object_id: `user-${newUser.id}`,
-            is_active: true,
-          }, { transaction: t });
-
-          // create token with provided user/address
-          await models.SsoToken.create({
-            issuer: userMetadata.issuer,
-            issued_at: user.claim.iat,
-            address_id: newAddress.id, // always ethereum address
-            created_at: new Date(),
-            updated_at: new Date(),
-          }, { transaction: t });
-
-          return newUser;
-        });
-
-        // re-fetch user to include address object
-        const newUser = await models.User.findOne({
-          where: {
-            id: result.id,
-          },
-          include: [ models.Address ],
-        });
-        return cb(null, newUser);
-      } else if (existingUser.Addresses?.length > 0) {
-        // each user should only ever have one token issued by Magic
-        const ssoToken = await models.SsoToken.findOne({
-          where: {
-            issuer: user.issuer
-          },
-          include: [{
-            model: models.Address,
-            where: { address: user.publicAddress }, // Assumes address is always Ethereum Address which it isn't
-            required: true,
-          }]
-        });
-        // login user if they registered via magic
-        if (user.claim.iat <= ssoToken.issued_at) {
-          console.log('Replay attack detected.');
-          return cb(null, null, {
-            message: `Replay attack detected for user ${user.publicAddress}}.`,
-          });
         }
-        ssoToken.issued_at = user.claim.iat;
-        ssoToken.updated_at = new Date();
-        await ssoToken.save();
-        console.log(`Found existing user: ${JSON.stringify(existingUser)}`);
-        return cb(null, existingUser);
-      } else {
-        // migrate to magic if email already exists
-        await sequelize.transaction(async (t) => {
-          const ethAddress = userMetadata.publicAddress;
-          const newAddress = await models.Address.create({
-            address: ethAddress,
-            chain: 'ethereum',
-            verification_token: 'MAGIC',
-            verification_token_expires: null,
-            verified: new Date(), // trust addresses from magic
-            last_active: new Date(),
-            user_id: existingUser.id,
-            profile_id: (existingUser.Profiles[0] as ProfileAttributes).id,
-            wallet_id: WalletId.Magic
-          }, { transaction: t });
-
-          await models.SsoToken.create({
-            issuer: userMetadata.issuer,
-            issued_at: user.claim.iat,
-            address_id: newAddress.id, // always ethereum address
-            created_at: new Date(),
-            updated_at: new Date(),
-          }, { transaction: t });
-        });
-        console.log(`Migrated old email user to magic: ${existingUser.email}`);
-        return cb(null, existingUser);
-      }
-    }));
+      })
+    );
```

Comment:
I have no idea what logical changes happened in this file. It looks as if the entire file was rewritten. Can you explain what changed, or modify the diff?

Reply:
Magic.ts when I merged in master had a syntax error. There was an extra curly brace that should not have been there. I made this change so that the app would be able to run

Reply:
fixed